compiler/rustc_middle/src/mir/query.rs

```diff
@@ -214,6 +214,7 @@ pub struct BorrowCheckResult<'tcx> {
     pub concrete_opaque_types: VecMap<OpaqueTypeKey<'tcx>, Ty<'tcx>>,
     pub closure_requirements: Option<ClosureRegionRequirements<'tcx>>,
     pub used_mut_upvars: SmallVec<[Field; 8]>,
+    pub tainted_by_errors: bool,
```

Comment:
Please make it an option like in the typeck results .this will make it easier for the in-flight PRs that make our error handling more solid